packages/apiserver/src/resources/materials/material.service.ts

```diff
@@ -103,4 +104,19 @@ export class MaterialService {
       throw new GraphQLError(e.message);
     }
   }
+  async getAllDonatedMaterialsByPerson(id: number | string) {
+    try {
+      return await Material.createQueryBuilder('material')
+        .leftJoinAndSelect('material.status', 'status')
+        .where('status.person_id = :person_id AND status.status LIKE :status', {
+          person_id: id,
```

Comment:
why is LIKE here. We are exactly a coincidence. It can also be broken down into 
.where('status.person_id = :person_id', {person_id: id}).
andWhere('status.status = :status', {status: "Pending"})